lib/plugins/keyGeneratorFactory.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+// TODO: should be instance! cannot use Nocca instance references now.
+
+var $q = require('q');
+var $extend = require('extend');
+
+module.exports = {
+    interface: 'keyGenerator',
+    id: 'keyGeneratorFactory',
+    name: 'Key Generator Factory',
+    constructor: KeyGeneratorFactory
+};
+
+function KeyGeneratorFactory (Nocca) {
+
+    var self = this;
+    
+    self.generateKey = generateKey;
+
+    self.incrementalKeyGenerator = incrementalKeyGenerator;
```

Comment:
oh shit, read over the combine part. 

you mean something like this in config?:
``` keyGenerator: ['keyGeneratorFactoryUrl', 'keyGeneratorFactoryBody']```

and then run all the plugins defined?